Update extractPersistedSchema for simple schema (#24562)
microsoft/FluidFramework: .changeset/tangy-times-carry.md

---
"fluid-framework": minor
"@fluidframework/tree": minor
"__section": tree
---

> The `extractPersistedSchema` `@alpha` API has its arguments adjusted

Reviewer: I'm slightly torn about this one. One of our guidelines for writing changesets suggests as a rule of thumb that "In this release, <heading>" should read well. The current heading works better in that structure, but the suggestion feels more natural to me. Maybe in part because it aligns with the "Remove <bla>" or "Deprecate <bla>" headings I think are common. Thoughts @tylerbutler , @jzaffiro , @Josmithr (or anyone else)?
```suggestion
Adjust the arguments of the `extractPersistedSchema` `@alpha` API
```

Reviewer 2: I've been trying to follow the documented guidance where possible. Personally, I think consistency is more important than occasional awkwardness in wording.

Reviewer 2: That said, it does seem like we could be a bit more specific. Maybe something like
> extractPersistedSchema (alpha) updated to accept SimpleTreeSchema input

Reviewer 2: Also, our current guidance asks us not to use `inlineCode` elements in the title. Fine in the body, but not in the title.

Author: How about:

```suggestion
The `extractPersistedSchema` `@alpha` API has had its arguments adjusted
```

I like keeping the impacted API at the beginning of the title (I think that makes skimming the release notes for relevant ones easier).

Reviewer 2: My suggestion above also keeps the impacted API at the beginning. Noting this in case your view of this thread wasn't updated when you replied 🙂

Author: I don't really like putting function names as the literal first word of a sentence since we can't capitalize them and that feels weird. Thats extra strange if we don't use inline code formatting.

Also your change doesn't address that it also has a second required argument now (if we are going to call out one argument change it seems odd to omit the other)

Author: @Josmithr you mention "our guidance". I checked `.changeset/README.md` which links 5 different documents, but not of them sound like they would have guidance for writing changesets. Where is this guidance, and how are we intended to discover it?

Reviewer 2: I had to do a bit of hunting, but these are the docs I was thinking of: https://eng.ms/docs/experiences-devices/opg/office-shared/fluid-framework/fluid-framework-internal/fluid-framework/docs/on-call/release/release-prep

Reviewer 2: We should probably move that over to `.changeset/README` as "Best Practices" documentation

Reviewer 2: I'll just do that now...
